### PR TITLE
Error 500 fix for Admin Categories If description null

### PR DESCRIPTION
**Pull request by uguranium**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Error fix for stripslashes(): Argument #1 ($string) must be of type string, null given. The parameter is null after migration. Please can we use the if condition and check if it's not null? Because if it is null gives a 500 error.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Error may happen after migration, to be tested by a dev
| Fixed ticket?     | n/a

### CODE REVIEW

**Review by Hlavtox — DISMISSED** [review]
I think it would be easier to cast it like:

`stripslashes((string) $record['description'])`

right @kpodemski?

**Comment by Hlavtox** [comment]
@uguranium Hi, could you please fill out the PR table? Thanks! :-)

**Comment by uguranium** [comment]
> stripslashes((string) $record['description'])

Yes it helps

**Comment by kpodemski** [comment]
@Hlavtox I like checking it `!== null` better 👍🏻

File: src/Core/Grid/Data/Factory/AbstractCategoryDataFactory.php [hunk 1]
```diff
@@ -74,7 +74,9 @@ public function getData(SearchCriteriaInterface $searchCriteria)
     protected function modifyRecords(array $records): array
     {
         foreach ($records as $key => $record) {
-            $records[$key]['description'] = mb_substr(strip_tags(stripslashes($record['description'])), 0, self::DESCRIPTION_MAX_LENGTH);
+            if($record['description'] !== null) {
```
Comment (kpodemski) [inline src/Core/Grid/Data/Factory/AbstractCategoryDataFactory.php]:
```suggestion
            if ($record['description'] !== null) {
```

coding standards

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
@uguranium @Hlavtox 

I edited the table and the code. It should be good to go. You can test it if you want @Hlavtox because the change is quite obvious.

**Review by matks — APPROVED** [review]

**Comment by kpodemski** [comment]
@PrestaShop/qa-team it's quite a technical change, we could use auto-tests and manual check by a dev, wdyt?

**Comment by paulnoelcholot** [comment]
@kpodemski I agree, QA can be done via auto-tests and manual test by a dev.

**Comment by kpodemski** [comment]
thank you @uguranium !

**Comment by ShaiMagal** [comment]
Guys, is possible to port this to 8.1 branch? We are facing this issue very often with customers on 8.1.x. Nearly second customer have this problem and we must fix it "manually".

Can we port this to 8.1 please as well?

@Hlavtox

**Comment by Hlavtox** [comment]
@ShaiMagal Yeah just make the PR with the same code, call it backport #30922 and reference the original author. ;-)

**Pull request merged**